src/components/token-equity-value/token-equity-value.tsx

```diff
@@ -20,17 +23,21 @@ interface Props {
 export const TokenEquityValue: FC<Props> = ({ token, exchangeRate }) => {
   const styles = useTokenEquityValueStyles();
 
+  const { hideBalanceHandler, isBalanceHidden } = useHideBalance();
+
   const formattedBalance = formatAssetAmount(new BigNumber(token.balance));
 
   return (
     <View style={styles.container}>
       <View style={styles.header}>
-        <Icon name={IconNameEnum.EyeOpenBold} size={formatSize(24)} />
+        <TouchableOpacity hitSlop={generateHitSlop(formatSize(10))} onPress={hideBalanceHandler}>
+          <Icon name={!isBalanceHidden ? IconNameEnum.EyeOpenBold : IconNameEnum.EyeClosedBold} size={formatSize(24)} />
+        </TouchableOpacity>
```

Comment:
Reuse `TouchableIcon`. 
and flip `Icon` `name` prop condition (it's easier to understand conditions without `!`)